piptools/repositories/pypi.py

```diff
@@ -125,18 +130,20 @@ def get_dependencies(self, ireq):
         if not (ireq.editable or is_pinned_requirement(ireq)):
             raise TypeError('Expected pinned or editable InstallRequirement, got {}'.format(ireq))
 
-        if not os.path.isdir(self._download_dir):
-            os.makedirs(self._download_dir)
-        if not os.path.isdir(self._wheel_download_dir):
-            os.makedirs(self._wheel_download_dir)
-
-        reqset = RequirementSet(self.build_dir,
-                                self.source_dir,
-                                download_dir=self._download_dir,
-                                wheel_download_dir=self._wheel_download_dir,
-                                session=self.session)
-        dependencies = reqset._prepare_file(self.finder, ireq)
-        return set(dependencies)
+        if ireq not in self._dependencies_cache:
```

Comment:
It might be better to key this cache off some derived key of ireq rather than using ireq itself, but I was unclear what would be the optimal key to use. Open to suggestions!